src/realm/object_id.cpp

```diff
@@ -88,6 +88,18 @@ ObjectId::ObjectId(Timestamp d, int machine_id, int process_id)
     m_bytes[11] = r & 0xff;
 }
 
+ObjectId::ObjectId(Timestamp d)
```

Comment:
I am not sure why we need this. If the ObjectId is expected to have a random number last, then this will not create a valid object.

Reply:
This PR needs to be rewritten, as Mathias points out, the this implementation of simply converting a Timestamp to an ObjectId like (as in the constructor here) will create some surprising behaviour when querying. The right thing to do will be to have official query support for ObjectId vs Timestamp, but that is going to take a significant amount of work, so it has been postponed given this feature is more of a "nice to have" at this point.